### PR TITLE
ci: Changing Service Connection for PR pipeline

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Moves the PR pipeline and all associated resources to the Build Validations Subscription to alleviate the resource group usage within Azure Network Agent - Test. 

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:
Added the service connection to the ACN-CNI-Pipeline Variable Group

### CODE REVIEW

**Review by paulyufan2 — COMMENTED** [review]
should we use all Build Validations Service connection for all tests? For example,. I can still see some tests are using "AZURE_TEST_AGENT_SERVICE_CONNECTION in: .pipelines/singletenancy/azure-cni-overlay/azure-cni-overlay-e2e-step-template.yaml

File: .pipelines/singletenancy/aks-swift/e2e-step-template.yaml
Comment (vipul-21) [inline .pipelines/singletenancy/aks-swift/e2e-step-template.yaml:1]: Do we want the CNI release pipeline( aka Load test pipeline) use the same azure sub as well ?

Comment (jpayne3506) [reply .pipelines/singletenancy/aks-swift/e2e-step-template.yaml:1]: This sounds reasonable. Added

**Review by paulyufan2 — DISMISSED** [review]
lgtm

**Review by vipul-21 — APPROVED** [review]

**Review by paulyufan2 — APPROVED** [review]
lgtm

**Pull request merged**